### PR TITLE
EVG-7515: realtime updates for patch build variants

### PR DESCRIPTION
**Pull request by tgrander**
Screen Shot 2020-04-29 at 6 51 45 PM

Patch build variants query polls the server until all tasks have ran or finished. 

I tested in staging.

### CODE REVIEW

File: cypress/integration/patch/patch-route/tasks-filters.ts [hunk 1]
```diff
@@ -75,7 +75,8 @@ const singularStatuses = [
   },
 ];
 
-describe("Tasks filters", function() {
+// TODO: refactor to not wait for / inspect gql query. Use defined data set and test against it.
```
Comment (khelif96) [inline cypress/integration/patch/patch-route/tasks-filters.ts:78]: Could you make a ticket for these so we dont forget to come back to them in the future.

File: src/pages/patch/BuildVariants.tsx [hunk 2]
```diff
@@ -12,15 +12,24 @@ import { Skeleton } from "antd";
 import { TaskSquare } from "./buildVariants/TaskSquare";
 import { H3, P1 } from "components/Typography";
 import styled from "@emotion/styled/macro";
+import every from "lodash/every";
+import { TaskStatus } from "types/task";
```
Comment (khelif96) [inline src/pages/patch/BuildVariants.tsx:16]: Does this type exist in the generated types?

File: cypress/integration/patch/patch-route/tasks-filters.ts (hunk 1 above)
Comment (tgrander) [reply cypress/integration/patch/patch-route/tasks-filters.ts:78]: one already exists. https://jira.mongodb.org/browse/EVG-7923

File: src/pages/patch/BuildVariants.tsx (hunk 2 above)
Comment (tgrander) [reply src/pages/patch/BuildVariants.tsx:16]: this is not a gql type and should never change, therefore does not need to be generated. it's a global evergreen type representing the different task statuses.

**Comment by tgrander** [comment]
failing tests = flakey my patches test

**Pull request merged**